elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryCleanerThread.java

```diff
@@ -0,0 +1,140 @@
+import com.yahoo.elide.async.models.QueryStatus;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.filter.dialect.RSQLFilterDialect;
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.request.EntityProjection;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Runnable thread for updating AsyncQueryThread status
+ * beyond the max run time and if not terminated by interrupt process
+ * due to app/host crash or restart.
+ */
+@Slf4j
+@Data
+@AllArgsConstructor
+public class AsyncQueryCleanerThread implements Runnable {
+
+    private int maxRunTimeMinutes;
+    private Elide elide;
+    private int queryCleanupDays;
+
+    @Override
+    public void run() {
+        deleteAsyncQuery();
+        timeoutAsyncQuery();
+    }
+
+    /**
+     * This method updates the status of long running async query which
+     * were not interrupted due to host crash/app shutdown to TIMEDOUT.
+     * */
+    @SuppressWarnings("unchecked")
+    private void deleteAsyncQuery() {
+        AsyncDbUtil asyncDbUtil = AsyncDbUtil.getInstance(elide);
+        Iterable<Object> loaded = (Iterable<Object>) asyncDbUtil.executeInTransaction(elide.getDataStore(), (tx, scope) -> {
+            try {
+                EntityDictionary dictionary = elide.getElideSettings().getDictionary();
+                RSQLFilterDialect filterParser = new RSQLFilterDialect(dictionary);
+
+                //Calculate date to clean up
+                Calendar cal = Calendar.getInstance();
+                cal.setTime(new Date());
+                cal.add(Calendar.DATE, -(queryCleanupDays));
+                Date cleanupDate = cal.getTime();
+                Format dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'");
+                String cleanupDateFormatted = dateFormat.format(cleanupDate);
+                log.debug("cleanupDateFormatted = {}", cleanupDateFormatted);
+
+                FilterExpression filter = filterParser.parseFilterExpression("createdOn=le='" + cleanupDateFormatted + "'",
+                        AsyncQuery.class, false);
+                log.debug("filter = {}", filter.toString());
+
+                EntityProjection asyncQueryCollection = EntityProjection.builder()
+                        .type(AsyncQuery.class)
+                        .filterExpression(filter)
+                        .build();
+
+                Iterable<Object> loadedObjects = tx.loadObjects(asyncQueryCollection, scope);
+                return loadedObjects;
+            } catch (Exception e) {
+                log.error("Exception: {}", e.getMessage());
+            }
+            return null;
+        });
+        Iterator<Object> itr = loaded.iterator();
+        while(itr.hasNext()) {
+            AsyncQuery query = (AsyncQuery) itr.next();
+
+            log.info("Found a query to DELETE");
+            asyncDbUtil.deleteAsyncQueryAndResult(query.getId());
```

Comment:
This is where using the DataStoreTransaction interface in Elide is less than ideal.  There is no bulk edit interface.  This will generate lots of round trips to the database to delete objects one at a time (rather than delete them in bulk).  We should at least change the DbUtil.deleteAsync function take a list of objects to delete - and to remove them in a single transaction.  DdUtil should be an interface that can be created and overridden (as a Spring bean) if there are performance problems with this general approach.